assets/js/linear_algebra/dot_product_symmetric2d.js

```diff
@@ -14,66 +14,27 @@ let origin = [150, 140],
-  return [
-      [
-          [
-              {text: ''}, {text: ''}, {text: ''}, {text: ''}, 
-              {text: 'v', text_opacity: 0.7, key: 'v'},
-          ], [
-              {text: ''}, {text: ''}, {text: ''}, {text: ''}, 
-              {text: v.coord.x.toFixed(2), text_opacity: 0.7, key: 'xv'},
-          ], [
-              {text: ''}, {text: ''}, {text: ''}, {text: ''},
-              {text: v.coord.z.toFixed(2), text_opacity: 0., tt: 0, key: 'zv'}
-          ], [
-              {text: ''}, {text: ''}, {text: ''}, {text: ''},
-              {text: v.coord.y.toFixed(2), text_opacity: 0.7, key: 'yv'}, 
-          ], [
-              {text: 'u\u1d40', text_opacity: 0.7, key: 'u'},
-              {text: u.coord.x.toFixed(2), text_opacity: 0.7, key: 'xu'},
-              {text: u.coord.z.toFixed(2), text_opacity: 0., tt: 0, key: 'zu'},
-              {text: u.coord.y.toFixed(2), text_opacity: 0.7, key: 'yu'},
-              {text: uTv.toFixed(3), font_size: 15, text_color: 0,
-               text_opacity: 1, key:'uTv'}
-          ]
-      ],
-      
-      [
-          [
-              {text: ''}, {text: ''}, {text: ''}, {text: ''}, 
-              {text: 'u', text_opacity: 0.7, key: 'u'}
-          ], [
-              {text: ''}, {text: ''}, {text: ''}, {text: ''}, 
-              {text: u.coord.x.toFixed(2), text_opacity: 0.7, key: 'xu'}
-          ], [
-              {text: ''}, {text: ''}, {text: ''}, {text: ''},
-
-              {text: u.coord.z.toFixed(2), text_opacity: 0., tt: 0, key: 'zu'},
-          ], [
-              {text: ''}, {text: ''}, {text: ''}, {text: ''},
-              {text: u.coord.y.toFixed(2), text_opacity: 0.7, key: 'yu'}
-          ], [
-              {text: 'v\u1d40', text_opacity: 0.7, key: 'v'},
-              {text: v.coord.x.toFixed(2), text_opacity: 0.7, key: 'xv'},
-              {text: v.coord.z.toFixed(2), text_opacity: 0., tt: 0, key: 'zv'},
-              {text: v.coord.y.toFixed(2), text_opacity: 0.7, key: 'yv'},
-              {text: uTv.toFixed(3), font_size: 15, text_color: 0,
-               text_opacity: 1, key:'uTv'}
-          ]
-      ]
-  ];
-}
-
+let row_width = 0.35,
+    col_width = 0.6,
+    w_unit = 1.0, h_unit = 1.0;
+
+    // dws_array = [0.5, 0.7, 0., 0.7],
+    // dhs_array = [0.4, 0.4, 0., 0.4];
+
+let start_coord_x=(340 - origin[0])/scale + 0.6 * w_unit,
+    start_coord_y=(75 - origin[1])/scale + 0.2 * h_unit,
+    last_col_coord = start_coord_x + 1.2 * w_unit,
+    last_row_coord = start_coord_y + 0.7 * h_unit;
+
+let v_name = {text: 'v =', x: last_col_coord,
+              y: start_coord_y - 0.4 * h_unit, key: 'v'},
+    u_name = {text: 'u =', x: last_col_coord,
+              y: start_coord_y - 0.4 * h_unit, key: 'u'},
+    vT_name = {text: 'v\u1d40 =', x: start_coord_x - 0.6 * w_unit,
+               y: last_row_coord + 0.05, key: 'v' },
+    uT_name = {text: 'u\u1d40 =', x: start_coord_x - 0.6 * w_unit,
+               y: last_row_coord + 0.05, key: 'u' };
```

Comment:
same request in 3D for these variable names

Reply:
done